Version 1.4.1

growdev opened the pull request:
Fix for version upgrade from v2 to v3 with different Form IDs.

#### Verify
 - [x] Page using global (default) form
 - [x] Page with specific form
 - [x] Page with landing page override
 - [x] Shortcode with no attributes
 - [x] Shortcode with id attribute
 - [x] Shortcode with form (version 2) attribute

marcboquet commented:
I'd add one more thing:
If the form endpoint returns `404` we should ignore it instead of printing the `{error: "not found"}`. A customer could delete a form from CK and we don't want this showing in their site.

marcboquet commented on wp-convertkit.php:
```diff
@@ -283,42 +285,138 @@ private static function _get_settings_page_link($query_args = array()) {
+				// TODO make v3 api call and return form markup
+
+
+			} else {
```
Maybe we could do an `elseif ( isset( $attributes['form'] ) )` here to handle the v2 shortcode, and handle the default_form in the `else`?

growdev commented:
@davidlamarwheeler @marcboquet This is ready to test again. You can checkout branch 1.3.9, setup your site, then checkout 1.4.1 to simulate upgrading. I test with these pages:

1. Page using global (default) form
2. Page with specific form
3. Page with landing page override
4. Shortcode with no attributes
5. Shortcode with `id` attribute
6. Shortcode with `form` (version 2) attribute

davidlamarwheeler commented:
Unfortunately, when I ran this upgrade locally, the last three in the checklist (at the top of this PR) didn't transfer correctly, and the Default Form didn't migrate.

Once I set the default form, it starts getting better, but it looks like the page with a specific form still didn't find its way to the correct ID.

davidlamarwheeler commented:
Okay, I did a second pass, after going into my db and removing the `convertkit_version` option and the empty `_wp_convertkit_upgrade_posts` option. Almost everything worked, except that one of my posts held on to the subscription_form ID. I'm trying to suss out why.

davidlamarwheeler commented on wp-convertkit.php:
```diff
@@ -281,48 +341,81 @@ private static function _get_settings_page_link($query_args = array()) {
+
+					// Done scanning posts, upgrade complete.
+					if ( empty( $posts ) )
+						update_option( 'convertkit_version', self::VERSION );
```
Should we also delete `_wp_convertkit_upgrade_posts` here?

davidlamarwheeler commented on wp-convertkit.php:
```diff
@@ -281,48 +341,81 @@ private static function _get_settings_page_link($query_args = array()) {
+							'fields' => 'ids',
+						);
+
+						$result = new WP_Query( $args );
```
I'm conspicuously only getting 10 results to loop through. Does this need a limit set/unset?

growdev commented:
@davidlamarwheeler I added a `posts_per_page` of `-1` to the query to get all the IDs. The default is 10, so that's why you were only getting 10.

I also removed the `_wp_convertkit_upgrade_posts` option when finished.

davidlamarwheeler commented:
Awesome! That worked perfectly. I think this looks ready to ship! :shipit:
